torch/_dynamo/optimizations/training.py

```diff
@@ -93,7 +93,14 @@ def compile_fn(cls, gm: torch.fx.GraphModule, example_inputs, **kwargs):
             return gm  # no point for tiny graphs
         return cls(gm, example_inputs, **kwargs).verified_candidate()
 
-    def __init__(self, gm: torch.fx.GraphModule, example_inputs, **kwargs):
+    def __init__(
+        self,
+        gm: torch.fx.GraphModule,
+        example_inputs,
+        fake_mode=None,
+        fake_inputs=None,
```

Comment:
Why do I need *both* example inputs and fake inputs?

Reply:
I'm not sure you should do this, but `fake_mode` memoizes conversions of the same real tensor into a fake tensor, so if you already called `from_real_tensor` on a tensor before, it will consistently give you back the same fake tensor (with the correct symints) in that case.

Reply:
And then, I guess, you could also technically just enable fake mode before calling the compilation function, and then you don't even need to do any of the kwargs business.

Reply:
You still need a way to pass in fake mode, so kwargs needs to be there.

There's also modes of operation for aot_autograd that take real tensors.